testsuite/MDAnalysisTests/analysis/test_encore.py

```diff
@@ -384,369 +368,341 @@ def test_ces_error_estimation_ensemble_bootstrap(self):
+                  [11.7470232, 10.60908885],
+                  [10.89279865, 9.32098256],
+                  [11.14254656, 9.28262927],
+                  [9.02660689, 9.12098876],
+                  [9.16093666, 8.72607596],
+                  [11.47151183, 8.92803007],
+                  [11.76917681, 9.59220592],
+                  [9.97880407, 11.26144744],
+                  [8.58057881, 8.43199283],
+                  [10.53394006, 9.36033059],
+                  [11.34577448, 10.70313399],
+                  [9.07097046, 8.83928763]])
+
+    XX = np.einsum('ij,ij->i', X, X)[:, np.newaxis]
+    YY = XX.T
+    distances = np.dot(X, X.T)
+    distances *= -2
+    distances += XX
+    distances += YY
+    np.maximum(distances, 0, out=distances)
+    distances.flat[::distances.shape[0] + 1] = 0.0
+
+    @pytest.fixture()
+    def dimension(self):
+        return len(self.distances)
+
+    @pytest.fixture()
+    def distance_matrix(self, dimension):
+        distance_matrix = encore.utils.TriangularMatrix(len(self.distances))
         for i in range(dimension):
-            for j in range(i,dimension):
-                self.distance_matrix[i, j] = distances[i,j]
+            for j in range(i, dimension):
+                distance_matrix[i, j] = self.distances[i, j]
+        return distance_matrix
 
-    def test_one(self):
-        preference = -float(np.median(self.distance_matrix.as_array()) * 10.)
+    def test_one(self, distance_matrix):
+        preference = -float(np.median(distance_matrix.as_array()) * 10.)
         clustering_method = encore.AffinityPropagationNative(preference=preference)
         ccs = encore.cluster(None,
-                             distance_matrix=self.distance_matrix,
+                             distance_matrix=distance_matrix,
                              method=clustering_method)
-        assert_equal(self.n_clusters, len(ccs),
-                     err_msg="Basic clustering test failed to give the right"
-                             "number of clusters: {0} vs {1}".format(self.n_clusters, len(ccs)))
-
-
-class TestEncoreDimensionalityReduction(TestCase):
-    def setUp(self):
-        # Create universe from templates defined in setUpClass
-        self.ens1 = mda.Universe(
-            self.ens1_template.filename,
-            self.ens1_template.trajectory.timeseries(format='fac'),
+        assert self.n_clusters == len(ccs), \
+                     "Basic clustering test failed to give the right"\
+                    "number of clusters: {0} vs {1}".format(self.n_clusters, len(ccs))
+
+
+class TestEncoreDimensionalityReduction(object):
+    @pytest.fixture(scope='class')
+    def ens1_template(self):
+        template = mda.Universe(PSF, DCD)
+        template.transfer_to_memory()
+        template = mda.Universe(
+            template.filename,
+            np.copy(template.trajectory.timeseries(format='fac')[::5, :, :]),
+                    format=mda.coordinates.memory.MemoryReader)
```

Comment:
I think `transfer_to_memory` has a step argument now that you can use instead of this trick.

Reply:
Why do you still not use the step argument of `transfer_to_memory` here?